storage/src/test/scala/io/delta/storage/integration/S3LogStoreUtilIntegrationTest.scala

```diff
@@ -0,0 +1,107 @@
+package io.delta.storage.integration
+
+import io.delta.storage.internal.S3LogStoreUtil
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.fs.Path
+import org.apache.hadoop.fs.s3a.{S3AFileSystem, UploadInfo}
+import org.scalatest.Tag
+import org.scalatest.funsuite.AnyFunSuite
+import org.scalactic.source
+
+import java.net.URI
+import java.nio.file.Paths
+import scala.math.max
+import scala.math.ceil
+import scala.math.round
+
+class S3LogStoreUtilIntegrationTest extends AnyFunSuite {
+  private val runIntegrationTests: Boolean =
+    Option(System.getenv("S3_LOG_STORE_UTIL_TEST_ENABLED")).exists(_.toBoolean)
+  private val bucket = System.getenv("S3_LOG_STORE_UTIL_TEST_BUCKET")
+  private val keyPrefix = System.getenv("S3_LOG_STORE_UTIL_TEST_KEY_PREFIX")
```

Comment:
what's the point of both `bucket` and `keyPrefix` by the way?
isn't `$bucket/$table/$filePath` sufficient?

Reply:
When testing this it is convenient to have a separate prefix for each test run. There are certainly multiple ways to go about this. We could for instance pick a random name for the tables, or let the prefix be part of the bucket environment variable.

Reply:
I see. can you please, then, add a comment to document this for `keyPrefix`, this wasn't immediately clear to me.

Reply:
`testRunPrefix` or `testRunUID` could also be a better name, btw

Reply:
Changed to testRunUID.